schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/communication/ApplicationContextImpl.java

```diff
@@ -0,0 +1,179 @@
+import org.apache.velocity.VelocityContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+
+/**
+ * @author krico
+ * @since 20/08/15.
+ */
+public class ApplicationContextImpl extends VelocityContext {
+    public static final String LOGO_PATH = "/build/img/jasify-logo-color.png";
+    private static final Logger log = LoggerFactory.getLogger(ApplicationContextImpl.class);
+
+    private static final TemplateEngine templateEngine = new TemplateEngineBuilder().build();
+
+    public ApplicationContextImpl() {
+        ApplicationContext.App app = createApp();
+        put(ApplicationContext.APP_KEY, app);
+        put(ApplicationContext.STRING_UTILS_KEY, new StringUtils());
+        put(ApplicationContext.KEY_UTIL_KEY, KeyUtil.INSTANCE);
+        put(ApplicationContext.FORMAT_UTIL_KEY, FormatUtil.INSTANCE);
+        put(ApplicationContext.CURRENCY_UTIL_KEY, CurrencyUtil.INSTANCE);
+        put(ApplicationContext.MODEL_UTIL_KEY, createModelUtil(app));
+        try {
+            put(ApplicationContext.STYLE_TOOL_KEY, templateEngine.getStyles(TemplateNames.STYLES_CSS));
+        } catch (TemplateEngineException e) {
+            log.error("FAiled to get styles", e);
+            throw Throwables.propagate(e);
+        }
+    }
+
+    protected ApplicationContext.App createApp() {
+        return new AppImpl();
+    }
+
+    protected ModelUtil createModelUtil(ApplicationContext.App app) {
+        return new ModelUtil(app);
+    }
+
+    public static final class AppImpl implements ApplicationContext.App {
+        public String getLogo() {
+            return getUrl() + LOGO_PATH;
+        }
+
+        public String getUrl() {
+            return EnvironmentUtil.defaultVersionUrl();
+        }
+    }
+
+    public static class ModelUtil {
+        private final ApplicationContext.App app;
+        private final ActivityDao activityDao = new ActivityDao();
+        private final ActivityPackageDao activityPackageDao = new ActivityPackageDao();
+        private final ActivityTypeDao activityTypeDao = new ActivityTypeDao();
+        private final OrganizationDao organizationDao = new OrganizationDao();
+        private final AttachmentDao attachmentDao = new AttachmentDao();
+        private final ActivityPackageSubscriptionDao activityPackageSubscriptionDao = new ActivityPackageSubscriptionDao();
+
+        ModelUtil(ApplicationContext.App app) {
+            this.app = app;
+        }
+
+        public String url(PasswordRecovery recovery) {
+            String code = recovery.getCode().getName();
+            GenericUrl recoverUrl = new GenericUrl(app.getUrl());
+            recoverUrl.clear();
+            recoverUrl.setRawPath("/");
+            recoverUrl.setFragment("/recover-password/" + code);
+            return recoverUrl.build();
+        }
+
+        public String name(User user) {
```

Comment:
I have seen a similar method in other code. Maybe a getDisplayName method in User to reduce copy paste code